model/labels/regexp.go

```diff
@@ -123,3 +137,55 @@ func optimizeConcatRegex(r *syntax.Regexp) (prefix, suffix, contains string) {
 
 	return
 }
+
+func findSetMatches(pattern string) (map[string]struct{}, string) {
```

Comment:
Maybe it is worth checking https://github.com/prometheus/prometheus/pull/13620.

I see same code path got modified

Reply:
Oh.. indeed very similar! :D That PR did not change the `FastRegexMatcher` but it included the same function! :D